client/app/views/steps/password.coffee

```diff
@@ -1,24 +1,63 @@
 StepView = require '../step'
 
+
 module.exports = class PasswordView extends StepView
 
     template: require '../templates/view_steps_password'
 
     events:
-        'click button': 'onSubmit'
+        'click .next': 'onSubmit'
+        'click [action=password-visibility]': 'onToggleVisibility'
+
+
+    isVisible: false
+
+
+    renderInput: =>
+        data = @serializeInputData()
+
+        # Show/hide password value
+        @$('input[name=password]').attr 'type', data.inputType
+
+        # Update Button title
+        @$('[action=password-visibility]').attr 'alt', t(data.visibilityTxt)
+
+        # Update Button Icon
+        @$('.icon use').attr 'xlink:href', data.visibilityIcon
 
 
     # Get 1rst error only
     # err is an object such as:
     # { type: 'password', text:'step empty fields'}
     serializeData: () ->
-        {
+        return Object.assign {}, @serializeInputData(), {
             error:      @error.message if @error
-            stepName:   @model.get 'name'
+            id:         "#{@model.get 'name'}-figure"
```

Comment:
Two questions for @m4dz here (not blocking for the current PR):

1. What do not use jade's region mechanism and template inheritance here ?
`id` and `figureid` don't need to be in Backbone View ? It is pure template logic, isn't it ?

2. Should we use a classname instead of an id for the figure name ? As we use it only for CSS styling.